refinery/core/models.py

```diff
@@ -1711,53 +1728,65 @@ def attach_outputs_dataset(self):
                     workflow_output=output_connection.name
                 )
             )
-            # retrieve uuid of corresponding output file if exists
-            logger.info("Results for '%s' and %s.%s: %s",
-                        self.uuid, output_connection,
-                        output_connection.filetype, analysis_result)
-
-            derived_data_file_node.file_uuid = analysis_result.file_store_uuid
-
-            logger.debug("Output file %s ('%s') assigned to node %s ('%s')",
-                         output_connection, analysis_result.file_store_uuid,
-                         derived_data_file_node.name,
-                         derived_data_file_node.uuid)
-
+            if output_connection.is_refinery_file:
+                # retrieve uuid of corresponding output file if exists
+                logger.info(
+                    "Results for '%s' and %s.%s: %s",
+                    self.uuid,
+                    output_connection,
+                    output_connection.filetype,
+                    analysis_result
+                )
+                derived_data_file_node.file_uuid = (
+                    analysis_result.file_store_uuid
+                )
+                logger.debug(
+                    "Output file %s ('%s') assigned to node %s ('%s')",
+                    output_connection,
+                    analysis_result.file_store_uuid,
+                    derived_data_file_node.name,
+                    derived_data_file_node.uuid
+                )
             output_connection.node = derived_data_file_node
             output_connection.save()
+
             # get graph edge that corresponds to this output node:
             # a. attach output node to source data transformation node
             # b. attach output node to target data transformation node
             # (if exists)
-            if len(graph.edges([output_connection.step])) > 0:
-                for edge in graph.edges_iter([output_connection.step]):
+            workflow_step = output_connection.step
+            if len(graph.edges([workflow_step])) > 0:
+                for edge in graph.edges_iter([workflow_step]):
                     output_id = "{}_{}".format(
-                        output_connection.step,
-                        output_connection.filename
+                        workflow_step,
+                        output_connection.name
                     )
                     if graph[edge[0]][edge[1]]['output_id'] == output_id:
-                        output_node_id = edge[0]
```

Comment:
**NOTE**: This is where things were flip-flopped before
This code wasn't broken, just difficult to debug since the naming didn't properly reflect what was happening. 

`output_node_id` was really the `input_node_id` etc.